### PR TITLE
edit course validation

### LINKED ISSUE #86: Edit course validation test

### PR DESCRIPTION
**Pull request by joel-foo**
closes #86

### CODE REVIEW

**Review by Carlintyj — COMMENTED** [review]
Hi Joel, great work on the edit function! I can really see the effort in making the code run efficiently. However, here are some nits to work on.

- Is it possible to have more extensive testing?
- Is there supposed to be an error message if I run "edit 1 c/NUR1113A-CS1101S" and NUR1113A is not a course that Person 1 is currently taking?
- Likewise is there supposed to be an error message if I run "delete 1 c/NUR1113A" and NUR1113A is not a course that Person 1 is currently taking?

Besides this, really great job on the code! Thanks for the effort.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -113,20 +113,21 @@ Examples:
...
-* The changes will be performed in the listed order.
-  * Assuming a specified student takes MA1521 initially:
-    * If the successive changes applied are MA1521-CS2100 and CS2100-MA1521, then MA1521 is still associated with the student.
-    * If the successive changes applied are MA1521-CS2103 and MA1521-CS2100, then MA1521 will be changed to CS2103 and not CS2100.
-* Editing a course will not result in duplicates. For example, CS2103T-MA1521 and CS2100-MA1521 will remove both CS2103T and CS2100, and create only one copy of MA1521.
+* To add a course, use c/add-[COURSE_TO_ADD].
+* To delete a course, use c/del-[COURSE_TO_DELETE].
+* To edit a course directly, use c/[ORIGINAL_COURSE-NEW_COURSE].
+* The three different types of course modifications can be performed in any amount and in any order. For example, it is possible to perform "c/add-MA1521 c/del-CS2103T c/MA2001-ST2334 c/add-CS2103T".
+* As usual, course validation will be performed. Any changes (and the changes thereafter) that fail course violation will not be executed.
```
Comment (chewjh1234) [inline docs/UserGuide.md:129]: I think you meant 'fail course validation' instead of 'fail course violation'.

**Review by chewjh1234 — APPROVED** [review]
I really love the edit function for courses. 

I have tried different invalid inputs and your code seems to work perfect! 
There is, I think, a small typo in the User Guide. 

Otherwise, LGTM!

**Comment by joel-foo** [comment]
> Hi Joel, great work on the edit function! I can really see the effort in making the code run efficiently. However, here are some nits to work on.
> 
> * Is it possible to have more extensive testing?
> * Is there supposed to be an error message if I run "edit 1 c/NUR1113A-CS1101S" and NUR1113A is not a course that Person 1 is currently taking?
> * Likewise is there supposed to be an error message if I run "delete 1 c/NUR1113A" and NUR1113A is not a course that Person 1 is currently taking?
> 
> Besides this, really great job on the code! Thanks for the effort.

Thank you! I will amend this in 1.3.

**Comment by joel-foo** [comment]
> I really love the edit function for courses.
> 
> I have tried different invalid inputs and your code seems to work perfect! There is, I think, a small typo in the User Guide.
> 
> Otherwise, LGTM!

Thank you! :)

**Pull request merged**